clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/PrepareTitusDeploy.java

```diff
@@ -0,0 +1,484 @@
+    if (description.getTargetGroups().isEmpty()) {
+      return null;
+    }
+
+    RegionScopedProviderFactory.RegionScopedProvider regionScopedProvider =
+        regionScopedProviderFactory.forRegion(
+            (NetflixAmazonCredentials)
+                accountCredentialsProvider.getCredentials(
+                    description.getCredentials().getAwsAccount()),
+            description.getRegion());
+
+    TargetGroupLookupHelper.TargetGroupLookupResult targetGroups =
+        targetGroupLookupHelper.getTargetGroupsByName(
+            regionScopedProvider, description.getTargetGroups());
+    if (!targetGroups.getUnknownTargetGroups().isEmpty()) {
+      throw new TargetGroupsNotFoundException(
+          format(
+              "Unable to find Target Groups: %s",
+              String.join(", ", targetGroups.getUnknownTargetGroups())));
+    }
+
+    return targetGroups;
+  }
+
+  @Nonnull
+  private Set<String> resolveSecurityGroups(Saga saga, TitusDeployDescription description) {
+    saga.log("Resolving Security Groups");
+
+    Set<String> securityGroups = new HashSet<>();
+    description
+        .getSecurityGroups()
+        .forEach(
+            providedSecurityGroup -> {
+              saga.log("Resolving Security Group '%s'", providedSecurityGroup);
+
+              if (awsLookupUtil.securityGroupIdExists(
+                  description.getAccount(), description.getRegion(), providedSecurityGroup)) {
+                securityGroups.add(providedSecurityGroup);
+              } else {
+                saga.log("Resolving Security Group name '%s'", providedSecurityGroup);
+                String convertedSecurityGroup =
+                    awsLookupUtil.convertSecurityGroupNameToId(
+                        description.getAccount(), description.getRegion(), providedSecurityGroup);
+                if (isNullOrEmpty(convertedSecurityGroup)) {
+                  throw new SecurityGroupNotFoundException(
+                      format("Security Group '%s' cannot be found", providedSecurityGroup));
+                }
+                securityGroups.add(convertedSecurityGroup);
+              }
+            });
+
+    saga.log("Finished resolving Security Groups");
+
+    return securityGroups;
+  }
+
+  @EqualsAndHashCode(callSuper = true)
+  @Value
+  public static class PrepareTitusDeployCommand extends SagaCommand
+      implements LoadFront50App.Front50AppAware {
+    private final TitusDeployDescription description;
+    @NonFinal private LoadFront50App.Front50App front50App;
+
+    public PrepareTitusDeployCommand(
+        @NotNull String sagaName, @NotNull String sagaId, TitusDeployDescription description) {
+      super(sagaName, sagaId);
+      this.description = description;
+    }
+
+    @Override
+    public void setFront50App(@Nonnull LoadFront50App.Front50App app) {
+      this.front50App = app;
+    }
+  }
+
+  private static class SecurityGroupNotFoundException extends TitusException {
```

Comment:
Feels like there could be a better way to share common exceptions for things that most cloud providers would need to use.

Reply:
So, yes I agree, but I think that Titus should still have an explicit exception for this. We don't have this ability yet, but being able to catch exceptions & then correlate them to a specific cloud provider would be immensely valuable for being able to track the reliability and correctness of a specific cloud provider.

Perhaps `clouddriver-core` could have a marker interface of `FirewallNotFoundException`, but I'd still want it as: `class SecurityGroupNotFoundException : TitusException, FirewallNotFoundException`

Does this make sense?